mozci/console/commands/decision.py

```diff
@@ -0,0 +1,116 @@
+
+        self.line(f"Process pushes from {branch}")
+
+        # List pushes
+        pushes = data.handler.get(
+            "pushes",
+            branch=branch,
+        )
+
+        for push in pushes:
+
+            # TODO: detect if a classification is already available
+            # and can be skipped
+
+            if dry_run:
+                self.line(f"Would classify {push.branch}@{push.rev}")
+                continue
+
+            # Create a child task to classify that push
+            task_id = self.create_task(push)
+            self.line(f"<info>Created task {task_id}</info>")
+
+    @memoized_property
+    def current_task(self):
+        """
+        Load the current task definition so that the children tasks
+        use the same image and workers
+        """
+        assert self.queue is not None, "Missing taskcluster queue"
+
+        task_id = os.environ.get("TASK_ID")
+        if not task_id:
+            raise Exception("Not in a Taskcluster environment")
+
+        task = self.queue.task(task_id)
+        task["id"] = task_id
+
+        return task
+
+    def create_task(self, push: Push) -> str:
+        """
+        Create a children task linked to the current task
+        that will classify a single push
+        """
+        task_id = taskcluster.slugId()
+        task = {
+            "created": taskcluster.stringDate(datetime.utcnow()),
+            "deadline": taskcluster.stringDate(taskcluster.fromNow("1 hour")),
+            "dependencies": [
+                self.current_task["id"],
+            ],
+            "metadata": {
+                "name": f"mozci classify {push.branch}@{push.rev}",
+                "description": "mozci classification task",
+                "owner": "mcastelluccio@mozilla.com",
+                "source": "https://github.com/mozilla/mozci",
+            },
+            "payload": {
+                "maxRunTime": 3600,
+                "image": self.current_task["payload"]["image"],
+                "command": [
+                    "push",
+                    "classify",
+                    push.branch,
+                    f"--rev={push.rev}",
+                    "--output=/tmp",
+                ],
+                "artifacts": {
+                    "public/classification.json": {
+                        "expires": taskcluster.stringDate(
+                            taskcluster.fromNow("3 months")
+                        ),
+                        "path": f"/tmp/classify_output_{push.branch}_{push.rev}.json",
+                        "type": "file",
+                    }
+                },
+            },
+            "routes": [
+                f"project.mozci.classification.{push.branch}.revision.{push.rev}",
+                f"project.mozci.classification.{push.branch}.push.{push.id}",
+            ],
+            "provisionerId": self.current_task["provisionerId"],
+            "workerType": self.current_task["workerType"],
+        }
```

Comment:
Do we need to define the group too, or is the new task going to be automatically in the same group as the decision task given the dependency?

Reply:
We do need to define it !